src/circular_buffer.jl

```diff
@@ -92,6 +92,12 @@ function Base.append!(cb::CircularBuffer, datavec::AbstractVector)
     cb
 end
 
+function Base.fill!(cb::CircularBuffer, data)
```

Comment:
This could do with a docstring.
Highlighting, exactly what it does.

In terms of grows the buffer up-to capacity, and fills it entirely.
Overwriting any existing elements.

Some might expect it to fill only unused elements.

Reply:
Should we have a keyword argument to distinguish the 2 kind of behavior?

What other collections are doing with `fill!`?

Reply:
`append!`  code could help